### PR TITLE
Resources page with routing and list of resource types

### PR DESCRIPTION
**Pull request by llorentelemmc**

### CODE REVIEW

File: AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceTypesRest.java [hunk 1]
```diff
@@ -49,13 +49,37 @@ public class ResourceTypesRest {
...
-    @ApiOperation(value = "Get all available ResourceTypes - used by Angular")
-    public List<ResourceTypeDTO> getAllResourceTypes() throws ValidationException {
+    @ApiOperation(value = "Get all resource types")
+    public List<ResourceTypeDTO> getAllResourceTypes() throws NotFoundException {
         List<ResourceTypeEntity> resourceTypes = resourceTypeLocator.getAllResourceTypes();
         List<ResourceTypeDTO> resourceTypeDTOs = new ArrayList<>();
         for (ResourceTypeEntity resourceType : resourceTypes) {
             resourceTypeDTOs.add(new ResourceTypeDTO(resourceType));
         }
         return resourceTypeDTOs;
```
Comment (mburri) [inline AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceTypesRest.java:59]: replace with stream().map()

File: AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceTypesRest.java [hunk 2]
```diff
@@ -49,13 +49,37 @@ public class ResourceTypesRest {
...
     }
+
+    @Path("/predefinedResourceTypes")
+    @GET
+    @ApiOperation(value = "Get predefined resource types")
+    public List<ResourceTypeDTO> getPredefinedResourceTypes() throws NotFoundException {
+        List<ResourceTypeEntity> resourceTypes = resourceTypeLocator.getPredefinedResourceTypes();
+        List<ResourceTypeDTO> resourceTypeDTOs = new ArrayList<>();
+        for (ResourceTypeEntity resourceType : resourceTypes) {
+            resourceTypeDTOs.add(new ResourceTypeDTO(resourceType));
```
Comment (mburri) [inline AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceTypesRest.java:69]: replace with stream().map()

File: AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceTypesRest.java [hunk 3]
```diff
@@ -49,13 +49,37 @@ public class ResourceTypesRest {
...
+    @Path("/rootResourceTypes")
+    @GET
+    @ApiOperation(value = "Get root resource types")
+    public List<ResourceTypeDTO> getRootResourceTypes() throws NotFoundException {
+        List<ResourceTypeEntity> resourceTypes = resourceTypeLocator.getRootResourceTypes();
+        List<ResourceTypeDTO> resourceTypeDTOs = new ArrayList<>();
+        for (ResourceTypeEntity resourceType : resourceTypes) {
+            resourceTypeDTOs.add(new ResourceTypeDTO(resourceType));
+        }
+        return resourceTypeDTOs;
```
Comment (mburri) [inline AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceTypesRest.java:83]: replace with stream().map()

**Review by mburri — CHANGES_REQUESTED** [review]

File: AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/boundary/ResourceTypeLocator.java [hunk 4]
```diff
@@ -43,8 +46,31 @@ public ResourceTypeEntity getResourceType(Integer resourceTypeId) {
...
+        for (ResourceTypeEntity e : resourceTypeDomainService.getResourceTypes()) {
+            if (e.getParentResourceType() == null && ResourceType.createByResourceType(e, null).isDefaultResourceType()) {
+                predefinedResourceTypes.add(e);
+            }
+        }
+        if (predefinedResourceTypes.isEmpty()) {
+            throw new NotFoundException("No predefined resource types found");
+        }
+        return predefinedResourceTypes;
+    }
```
[the remark this answers is not in the document]
Comment (mburri) [reply AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/boundary/ResourceTypeLocator.java:68]: `GET /thing/:id` should return 404 if the thing with the given id was not found
`GET /thing` should **not** return 404 if the list is empty

File: AMW_angular/io/src/app/resources/resources-page.component.html [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    </nav>
+    <nav class="nav flex-column">
+      @for (resourceType of rootResourceTypes(); track resourceType.name) {
+      <a (click)="toggleChildren(resourceType)">
+        {{ resourceType.name }}
+        @if (resourceType.hasChildren) {
+        <span>{{ expandedResourceTypeId === resourceType.id ? '-' : '+' }}</span>
+        }
+      </a>
+      @if (expandedResourceTypeId === resourceType.id) {
```
Comment (BeriBoss) [inline AMW_angular/io/src/app/resources/resources-page.component.html:19]:
```suggestion
      @if (resourceType.hasChildren && expandedResourceTypeId === resourceType.id) {
```

**Pull request merged**